OGC Charter (#23)
zarr-developers/geozarr-spec: CHARTER.adoc

:toc:
:toc-placement!:
:toclevels: 4
:numbered:
:sectanchors:
:source-highlighter: pygments

<<<
[cols = ">",frame = "none",grid = "none"]
|===
|{set:cellbgcolor:#FFFFFF}
|[big]*Open Geospatial Consortium*
|Submission Date: 2023-05-01
|Approval Date: 
|Internal reference number of this OGC(R) document: 23-xxx
|Category: OGC(R) Standards Working Group Charter
|Authors: Christophe Noel, Brianna R. Pagán
|===

[cols = "^", frame = "none"]
|===
|[big]*{titletext}*
|===

[cols = "^", frame = "none", grid = "none"]
|===
|*Copyright notice*
|Copyright (C) 2021 Open Geospatial Consortium
|To obtain additional rights of use, visit http://www.opengeospatial.org/legal/
|===

<<<

To: OGC members & interested parties

A new OGC Standards Working Group (SWG) is being formed. The OGC members listed below have proposed the OGC GeoZarr SWG.  The SWG proposal provided in this document meets the requirements of the OGC Technical Committee (TC) Policies and Procedures.

The SWG name, statement of purpose, scope, list of deliverables, audience, and language specified in the proposal will constitute the SWG's official charter. Technical discussions may occur no sooner than the SWG's first meeting.

> This SWG will operate under the OGC IPR Policy. The eligibility requirements for becoming a participant in the SWG at the first meeting (see details below) are that:

Reviewer: This text is standard letter of the OGC (see for example https://portal.ogc.org/files/96156), so I don't know if we can change this part.

* You must be an employee of an OGC member organization or an individual

> member of OGC;

Reviewer 2: I'm pretty uncomfortable with this requirement. Can we say instead that any interested party can join the swg?

Reviewer: No that's not possible. However, Scott already explained how interested parties can attend the meetings and contribute after signing an administrative paper.

Reviewer 3: Voting SWG members must be OGC members but OGC activities can be inclusive such that a person is a SWG contributor. Perhaps clarifying the distinction between member and contributor would be useful?

Reviewer: This text is standard letter of the OGC (see for example https://portal.ogc.org/files/96156), so I don't know if we can change this part.

* The OGC member must have signed the OGC Membership agreement;

* You must notify the SWG chair of your intent to participate to the first meeting. Members may do so by logging onto the OGC Portal and navigating to the Observer page and clicking on the link for the SWG they wish to join and;

* You must attend meetings of the SWG. The first meeting of this SWG is at the time and date fixed below. Attendance may be by teleconference.

Of course, participants also may join the SWG at any time. The OGC and the SWG welcomes all interested parties.

Non-OGC members who wish to participate may contact us about joining the OGC. In addition, the public may access some of the resources maintained for each SWG: the SWG public description, the SWG Charter, Change Requests, and public comments, which will be linked from the SWG’s page.

Please feel free to forward this announcement to any other appropriate lists. The OGC is an open standards organization; we encourage your feedback.

== Purpose of the Standards Working Group

> The GeoZarr Standard Working Group (SWG) is chartered to develop a Zarr encoding for geospatial gridded data in the form of Zarr conventions (based on the approach described in the draft Zarr Enhancement Proposal 4).  Zarr specifies a protocol and format used for storing Zarr arrays, while GeoZarr defines **conventions** and recommendations for storing **multidimensional georeferenced grid** of geospatial observations (including rasters). The GeoZarr SWG will also work on improving the Climate and Forecast (CF) conventions if necessary, particularly for alternative coordinate reference system (CRS) encoding if relevant.

Reviewer 4: > The GeoZarr SWG will also work on improving the Climate and Forecast (CF) conventions if necessary, particularly for alternative coordinate reference system (CRS) encoding if relevant.

Not sure we want to officially commit to revising CF conventions. Consider relaxing this phrasing to something like:

> If appropriate, the GeoZarr SWG will **develop proposed changes** to the CF conventions...

Reviewer 5: I agree and would suggest working with other CF groups in developing proposed changes. So perhaps:

> If appropriate, the GeoZarr SWG will, working with the OGC netCDF SWG and the CF community, develop proposed changes to the CF conventions ...

The OGC netCDF SWG is responsible for the OGC CF and netCDF documents. The [CF community governance process](https://cfconventions.org/governance.html) is how changes and enhancements are made to the CF specification.

Reviewer: I agree also

Reviewer: ```suggestion
The GeoZarr Standard Working Group (SWG) is chartered to develop a Zarr encoding for geospatial gridded data in the form of Zarr conventions (based on the approach described in the draft Zarr Enhancement Proposal 4).  Zarr specifies a protocol and format used for storing Zarr arrays, while GeoZarr defines **conventions** and recommendations for storing **multidimensional georeferenced grid** of geospatial observations (including rasters). If appropriate, the GeoZarr SWG may also contribute to improving the Climate and Forecast (CF) conventions (e.g. alternative CRS encoding), then propose changes following the [CF community governance process](https://cfconventions.org/governance.html).
```

> The GeoZarr Standard Working Group (SWG) is chartered to develop a Zarr encoding for geospatial gridded data in the form of Zarr conventions (based on the approach described in the draft Zarr Enhancement Proposal 4).  Zarr specifies a protocol and format used for storing Zarr arrays, while GeoZarr defines **conventions** and recommendations for storing **multidimensional georeferenced grid** of geospatial observations (including rasters). The GeoZarr SWG will also work on improving the Climate and Forecast (CF) conventions if necessary, particularly for alternative coordinate reference system (CRS) encoding if relevant.

Reviewer: added in references

== Business Value Proposition

> In the geospatial world, new cloud native data formats are emerging. Zarr is a generic data format for n-dimensional arrays that enables access to data in compressed chunks of the original array and has become increasingly popular to use for geospatial purposes. In June 2022, the OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html). The purpose of this charter is to adopt a more explicit GeoZarr as an OGC Standard which would provide guidance to standardize the approach for encoding various aspects of geospatial data in zarrs.

Reviewer: ```suggestion
In the geospatial world, new cloud native data formats are emerging. Zarr is a generic data format for n-dimensional arrays that enables access to data in compressed chunks of the original array and has become increasingly popular to use for geospatial purposes. Zarr facilitates portability and interoperability on both object stores and hard disks. In June 2022, the OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html). The purpose of this charter is to adopt a more explicit GeoZarr as an OGC Standard which would provide guidance to standardize the approach for encoding various aspects of geospatial data in Zarr stores. 
```

== Scope of Work

The goal of the GeoZarr specification is to establish flexible and inclusive conventions for the Zarr cloud-native format, specifically designed to meet the diverse requirements within the geospatial domain. These conventions aim to provide a clear and standardized framework for organizing and describing data, ensuring unambiguous representation. 

The objectives of GeoZarr conventions includes:

1. Compatibility: Ensuring easy compatibility with popular mapping and data analysis tools such as GDAL, Xarray, ArcGIS, and QGIS, enabling seamless integration into existing workflows.
2. Visualisation: Simplifying the creation and display of geospatial data in web browsers without the need for complex workarounds, making geospatial information more accessible to users.
3. Dimensions: Supporting multidimensional data, such as hyperspectral and altitude information, to address diverse geospatial data requirements.

> 4. Data Discovery: Providing metadata for discovering, accessing, and retrieving the data, including subsets or different arrangements of the data.

Reviewer: ```suggestion
4. Data Discovery: Providing metadata for discovering, accessing, and retrieving the data, including composite products made of multiple data arrays .
```

5. Mixing Data: Facilitating the combination of different types of geospatial data, including satellite images, elevation maps, and weather models, to create comprehensive and informative datasets.

> 8. Flexibilty: Allowing scientists and researchers to work with diverse data types and projections in their preferred software and programming languages, promoting flexibility and adaptability in geospatial data processing and analysis.

Reviewer 4: I fought (and lost) the battle to explicitly _exclude_ visualization from the OGC GeoDataCube SWG charter scope. But I'm going to try again!

In my opinion, everything after objective (1) is downstream of objective (1). I.e., I might consider simplifying this to something like:

> The overarching objective of the GeoZarr conventions is to provide a single unified specification for spatially-referenced data that simultaneously supports existing tooling and workflows for working with multi-dimensional gridded data (e.g., Xarray, NetCDF libraries) and existing tooling and workflows for efficient analysis if georeferenced raster and vector data (e.g., GDAL). This will enable GeoZarr datasets to be seamlessly integrated into existing workflows and tools for interactive analysis and visualization by a variety of users.

Reviewer: I strongly disagree, as it is a key objective of cloud-native data formats (similar as COG). Furthermore, achieving this objective does not require significant effort or complexity.

Reviewer 3: I would support keeping visualization in scope. Visual interpretation of these data is fundamental to any development activity and should be a first class use case.

Author: I have taken out visualization as it's own item, instead nodding to it under compatibility. Also I have changed the wording of the following paragraph to use the terminology conformance classes. I think this will get us un-stuck and allow for further discussions within the SWG.  

> Specifically, the convention should provide guidance to standardize the approach for encoding various aspects of geospatial data, including, for example the following list of potential conformance classes: 

> Part of the effort of this working group is to determine what should be kept as a part of the GeoZarr core requirements, and what aspects of geospatial data should be kept as separate conformance classes, which may or may not be applicable to specific sub-domains of the geospatial community.

> Specifically, the convention should provide guidance to standardize the approach for encoding various aspects of geospatial data, including:

Reviewer 2: ```suggestion
Specifically, the convention should provide guidance to standardize the approach for encoding various aspects of geospatial data, including, for example:
```

* Multiple related variables with heterogeneous coordinates (e.g., children or linked datasets)

> * Multiple resolutions of the data, possibly leveraging multiscale drafts

Reviewer 2: ```suggestion
* Multiple resolutions of the data, possibly leveraging multiscale array representations
```

Reviewer: ```suggestion
* Multiple resolutions of the data, as proposed in the GeoZarr draft specfication (multiscale property)
```

> * Multiple resolutions of the data, possibly leveraging multiscale drafts

Reviewer 4: Seems like a separate issue --- suggest removing as out-of-scope (or saying something like, "Support for / compatibility with Zarr extensions, such as multi-resolution (LINK)."

Reviewer 6: @ashiklom I'd suggest we keep `Multiple related variables with heterogeneous coordinates (e.g., children or linked datasets)` so we accurately capture the need to encode potentially different CRS information for different variables.

As you and @rabernat discussed, multi-resolution is not specific to domain so let's move this concept to a new Zarr extension.

Reviewer: Disagreed. 
The inclusion of multiscales is a crucial aspect in enabling fast visualization (as for COG). 
In the geospatial domain, specific attributes may be utilized for describing multiscales (as outlined in the draft), and it would typically be linked to OSM levels.

The specification should  encompass all pertinent aspects instead of excluding potentially more generic ones. If a candidate convention emerges that adequately fulfills all needs, it can then be updated to a reference within the specification.

Reviewer 3: It can be useful to develop these approaches with a specific use case in mind. I'd suggest leaving this in scope here but planning to work the problem in parallel in a more general context.

Author: I am bringing this up for discussion at our bi-weekly meeting. It's just one of a few examples where the need is actually upstream, but right now there is no champion for implementation.

Reviewer 7: We just talked about this at the meeting, and multi-resolution (and for that matter multiple chunking schemes) sound like they should be their own (optional) conformance classes.  COG is a good example to look at where overviews are a separate, thus optional, conformance/requirements class.

Reviewer: @briannapagan , @matthewhanson  : I have the same feeling (conformance classes for multi-resolution and a large set of other topics). However, I believe this is not the right time to discuss those aspect: when draft covnentions will be ready for a large set of topics, then we will be able to discuss how make a consistent spec.

Author: As in previous resolved conversation I have added the lines - which I hope is a good middle ground for all:

> Specifically, the convention should provide guidance to standardize the approach for encoding various aspects of geospatial data, including, for example the following list of potential conformance classes:

> Part of the effort of this working group is to determine what should be kept as a part of the GeoZarr core requirements, and what aspects of geospatial data should be kept as separate conformance classes, which may or may not be applicable to specific sub-domains of the geospatial community.

* Data subsets that are only available at certain resolutions

> * Multiple projections

Reviewer 2: ```suggestion
```

Removing this because I don't understand what is meant here. Do we want to store multiple projections of the same data? That feels like unnecessary scope creep.

Reviewer: That's maybe not a major/primordial aspect but as some big data cubes will be migrated to cloud-native data (for acting like data access and map tile services), it will typically store in the zarr store data in a world-wide projection, and in a local projection (use to fasten processing / visualisation of the data).

Indeed, it is important to identify secondary aspects that may not be immediately addressed in the initial version.

Author: Removing per Ryan's suggestion

> * Multiple projections

Reviewer 4: ```suggestion
* Multiple projections within a single Zarr store
```

Reviewer: Not necessarily.

Reviewer 3: That point should be clarified. As is, "within a single Zarr store" is kind of implied.

Reviewer: I don't share your point. The convention should provide guidance on handling multiple project: the solution may rely on stating they should be stored in individual Zarr or specify a structure (e.g. subgroups) and property.

Reviewer: Maybe ?
```suggestion
* Multiple projections (subgroups, individual zarr stores, ...)
```

> * Multi-dimensional optimizations (rechunking)

Reviewer 5: Like multiple resolutions, support for multiple chunking schemes is not specific to the geospatial domain. If multiple resolutions is moved into a separate Zarr convention, would it make sense to move chunking into that (or its own) convention as well.

It also feels like support for multiple chunking schemes, multiple resolutions, and multiple projections might all have some commonality as they are all trying to describe that two arrays represent the same data with differences in how the two arrays are stored.

Reviewer: If a candidate convention emerges that adequately fulfills all needs, it can then be updated to a reference within the specification.
Geospatial specific convention might for example defines the two most common dimensions for which optimized chunkings are available.
It would also specify in which case this would be relevant (e.g. for XYT(W) dimensions)

Reviewer: ```suggestion
* Multi-dimensional optimizations (spatial chunking scheme, temporal chunking scheme)
```

> * Supporting typical Earth observation (EO) products (for example, how to encode multispectral bands)

Reviewer 2: Do CF conventions address hyperspectral data? I don't know why we have to call out hyperspectral specifically. It seems like one of very many different applications.

In general, my preference is to put as little application-specific detail as possible into GeoZarr and defer to existing conventions (e.g. CF, GDAL) wherever possible.

Reviewer: We have already stored hyperspectral data in Zarr in HDSA (as it was a key objective).

For multispectral/hyperspectral data, it is not only a question of is there any metadata/convention provided in CF/GDAL: it is more about standardizing the way/structure/... it is encoded. (I guess with all benefits of standardization).

Not going into details, it might be conventions such as:
- Various bands must be encoded in the same data array using a wavelength dimension which value provided the associated spectral wavelength. 
- If subsets are encoded in different dimensions, the subset are encoded in individual subgroups (or zarr ?)
- etc.

Reviewer 3: The metadata convention to allow unambiguous automated interpretation of multiband or more sophisticated (different pixel size per band) is not part of CF even though some rudimentary support may be there in standard names. I support keeping this in scope.

> * Encoding of time series that have not been normalized

Reviewer 2: I don't understand what is meant here either. This feels too specific to put in the charter.
```suggestion
```

Reviewer: Not exactly. This covers the very usual use case of series of observations for which the footprint/pixels are not aligned but covering the same area. 
So can we provide a convention to encode this as a single array (with time dimension) and how (and let know the reader).

Reviewer 3: Consider expanding the definition to "Encoding time series of images that share a footprint but not grid." 

That said, this is a very boutique data format that may not be worth the energy to standardize in a single ZARR structure.

Author: I am also in favor of removing, i am not sure we can generalize any convention enough to cover all use cases. It is indeed very common, but you often have additional dimensions that allow representation as a single array. Lots of level 2 products do this.

Reviewer: Being not sure... means this question should be addressed by the WG and thus, present in the charter.  :)

> * Accessing the symbology of the corresponding data

Reviewer: Symbology provides indication (not impose) about how this particular data can be displayed (for example following NDVI symbology if it is symbology).

> * Accessing the symbology of the corresponding data

Author: I am also hesitant on adding symbology as part of a convention. This feels too specific.

> * Accessing the symbology of the corresponding data

Author: As these have been stored in STAC, can leave in for consideration and discussion as conformance class

=== Statement of relationship of planned work to the current OGC standards baseline

> As the existing draft GeoZarr metadata utilizes Climate and Forecast (CF) attributes, there is an expected relationship with the OGC CF-netCDF Data Model Extension Standard. If necessary, any adjustments or enhancements required for the GeoZarr specification will be considered as a proposal to improve CF conventions as well.

Reviewer 4: Suggest adding:

> The GeoZarr specification is also relevant to the OGC GeoDataCube SWG, for which it may become a candidate format.

Reviewer: A backend format ?
As GeoDataCube is an API, I don't think it's particularly suiting well.

Author: We have added a reference of GeoDataCube to 8.2. Similar or Applicable Standards Work (OGC and Elsewhere)

> As the existing draft GeoZarr metadata utilizes Climate and Forecast (CF) attributes, there is an expected relationship with the OGC CF-netCDF Data Model Extension Standard. If necessary, any adjustments or enhancements required for the GeoZarr specification will be considered as a proposal to improve CF conventions as well.

Reviewer 5: Adding mention of the OGC netCDF SWG.

```suggestion
As the existing draft GeoZarr metadata utilizes Climate and Forecast (CF) attributes, there is an expected relationship with the OGC CF-netCDF Data Model Extension Standard and the OGC netCDF SWG. If necessary, any adjustments or enhancements required for the GeoZarr specification will be considered as a proposal to improve CF conventions as well.
```

> 

Reviewer 2: ```suggestion
There are strong connections to other OGC raster / array container standards, specifically NetCDF, HDF5, and GeoTiff. The GeoZarr SWG should seek to leverage existing metadata conventions wherever possible.

As a chunked storage format, there is potentially a strong connection to the `OGC API - Tiles` standard. Map tiles could essentially be mapped 1:1 to an appropriately defined multiscale Zarr array.
```

=== What is Out of Scope?

> In early conversations around creating a draft GeoZarr specification, concerns arose multiple times around the CF encoding of CRS which may pose issues, see https://github.com/zarr-developers/geozarr-spec/issues/20. While these concerns will be discussed and suggestions created for potentially updating CF conventions, if resolutions cannot be made with the GeoZarr specification, we consider out of scope waiting on any subsequent updates to CF convetions to reflect these suggestions. 

Reviewer 5: This seems counter to the sentence (and discussion) above in the Purpose section:
> The GeoZarr SWG will also work on improving the Climate and Forecast (CF) conventions if necessary, particularly for alternative coordinate reference system (CRS) encoding if relevant.

I agree GeoZarr shouldn't wait forever for changes to CF. However, working with the CF community for some reasonable amount of time certainly seems within scope for GeoZarr.

Perhaps this should be removed and some mention of not being totally blocked by CF disagreements should be added in the Purpose section or the Relationshop with Other OGC Standards section?

Not sure what should be out of scope. Perhaps something generic and simple like:
> The GeoZarr SWG will only consider matters related to encoding geospatial data in Zarr.

=== Additional SWG Tasks

No specific additional tasks are currently planned for the SWG.

== IPR Policy for this SWG

[x] RAND-Royalty Free

[ ] RAND for fee

== Anticipated Audience / Participants

This SWG will develop a Standard for general use in the geospatial community and suitable for data exchange beyond this community. Geospatial data providers and software implementers will be interested in assisting with the development of this Standard as well as the output of the SWG.

== Domain Working Group Endorsement

The SWG convenors will discuss the charter with potentially interested Domain Working Groups (DWGs) at the first opportunity.

== Other informative information about the work of this SWG

=== Collaboration

All work in the Standards Working Group will be public and the SWG solicits contributions and feedback from OGC members and non-OGC members to the extent that is supported by the OGC Technical Committee Policies and Procedures.

The OGC GeoZarr SWG will collaborate on Standard development using a public GitHub repository and a Gitter channel. Development of the Standard will include the use of Issues and other project tools in GitHub.

> === Similar or Applicable Standards Work (OGC and Elsewhere)

Reviewer 4: Add OGC GeoDataCube SWG here.

> * The OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html) in June 2022.

Reviewer 2: ```suggestion
* The OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html) in June 2022.

* This SWG is closely related to the newly announced [Geodatacube SWG](https://www.ogc.org/press-release/ogc-forms-new-geodatacube-standards-working-group/). Essentially, GeoZarr will define a standard for a serverless datacube. Therefore, close coordination between these SWGs seems needed.
```

> The XCube project has potential synergies with the GeoZarr specification as it already relies and complies with CF conventions: 

Reviewer 2: ```suggestion
* The XCube project has potential synergies with the GeoZarr specification as it already relies and complies with CF conventions: 
```

The SWG name, statement of purpose, scope, list of deliverables, audience, and language specified in the proposal will constitute the SWG's official charter. Technical discussions may occur no sooner than the SWG's first meeting.

This SWG will operate under the OGC IPR Policy. The eligibility requirements for becoming a participant in the SWG at the first meeting (see details below) are that:

* You must be an employee of an OGC member organization or an individual
member of OGC;

* The OGC member must have signed the OGC Membership agreement;

* You must notify the SWG chair of your intent to participate to the first meeting. Members may do so by logging onto the OGC Portal and navigating to the Observer page and clicking on the link for the SWG they wish to join and;

* You must attend meetings of the SWG. The first meeting of this SWG is at the time and date fixed below. Attendance may be by teleconference.

Of course, participants also may join the SWG at any time. The OGC and the SWG welcomes all interested parties.

> Non-OGC members who wish to participate may contact us about joining the OGC. In addition, the public may access some of the resources maintained for each SWG: the SWG public description, the SWG Charter, Change Requests, and public comments, which will be linked from the SWG’s page.

Reviewer: See https://portal.ogc.org/files/96156 . Text seems part of every charter.

== Business Value Proposition

In the geospatial world, new cloud native data formats are emerging. Zarr is a generic data format for n-dimensional arrays that enables access to data in compressed chunks of the original array and has become increasingly popular to use for geospatial purposes. In June 2022, the OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html). The purpose of this charter is to adopt a more explicit GeoZarr as an OGC Standard which would provide guidance to standardize the approach for encoding various aspects of geospatial data in zarrs.

== Scope of Work

The goal of the GeoZarr specification is to establish flexible and inclusive conventions for the Zarr cloud-native format, specifically designed to meet the diverse requirements within the geospatial domain. These conventions aim to provide a clear and standardized framework for organizing and describing data, ensuring unambiguous representation. 

The objectives of GeoZarr conventions includes:

> 1. Compatibility: Ensuring easy compatibility with popular mapping and data analysis tools such as GDAL, Xarray, ArcGIS, and QGIS, enabling seamless integration into existing workflows.

Reviewer: It doesn't sound correct to talk about compatibility. Many formats can be mapped to Zarr encoding.

* Multi-dimensional optimizations (rechunking)
* Supporting typical Earth observation (EO) products (for example, how to encode multispectral bands)
* Encoding of time series that have not been normalized
* Accessing the symbology of the corresponding data

=== Statement of relationship of planned work to the current OGC standards baseline
As the existing draft GeoZarr metadata utilizes Climate and Forecast (CF) attributes, there is an expected relationship with the OGC CF-netCDF Data Model Extension Standard. If necessary, any adjustments or enhancements required for the GeoZarr specification will be considered as a proposal to improve CF conventions as well.

=== What is Out of Scope?
In early conversations around creating a draft GeoZarr specification, concerns arose multiple times around the CF encoding of CRS which may pose issues, see https://github.com/zarr-developers/geozarr-spec/issues/20. While these concerns will be discussed and suggestions created for potentially updating CF conventions, if resolutions cannot be made with the GeoZarr specification, we consider out of scope waiting on any subsequent updates to CF conventions to reflect these suggestions. 

> 

Reviewer: I disagree the charter should determine that. I also disagree about descoping those topics (which are covered in the original draft).

=== Specific Existing Work Used as Starting Point
* GeoZarr draft specification: https://github.com/zarr-developers/geozarr-spec/ 

=== Is This a Persistent SWG

[x] YES

[ ] NO

=== When can the SWG be Inactivated

The SWG can be inactivated once the SWG identifies no new tasks for the SWG and there are no open Change Requests.

== Description of deliverables
The GeoZarr SWG will deliver a candidate Standard and associated developer resources.

> The SWG expects to have a candidate Standard ready for OGC Architecture Board (OAB) review and public comment within nine months of creation of the SWG. Because example implementations will be developed at the same time the candidate Standard is formalized, reference implementations that fully use GeoParquet should be documented at the same time the candidate Standard goes to vote.

Author: This was meant to say GeoZarr updated.

=== Initial Deliverables

The following deliverables will be the initial results of work of the SWG.

* OGC GeoZarr Standard

* GeoZarr developer resources

The targeted start date for this SWG is 

=== Additional SWG Tasks

No specific additional tasks are currently planned for the SWG.

== IPR Policy for this SWG

[x] RAND-Royalty Free

[ ] RAND for fee

== Anticipated Audience / Participants

This SWG will develop a Standard for general use in the geospatial community and suitable for data exchange beyond this community. Geospatial data providers and software implementers will be interested in assisting with the development of this Standard as well as the output of the SWG.

== Domain Working Group Endorsement

> The SWG convenors will discuss the charter with potentially interested Domain Working Groups (DWGs) at the first opportunity.

Author: Very little feedback from DWG - hopefully better luck in September with charter submitted and on agendas.

== Other informative information about the work of this SWG

=== Collaboration

All work in the Standards Working Group will be public and the SWG solicits contributions and feedback from OGC members and non-OGC members to the extent that is supported by the OGC Technical Committee Policies and Procedures.

The OGC GeoZarr SWG will collaborate on Standard development using a public GitHub repository and a Gitter channel. Development of the Standard will include the use of Issues and other project tools in GitHub.

=== Similar or Applicable Standards Work (OGC and Elsewhere)

* The OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html) in June 2022.

> * This SWG is closely related to the newly announced [Geodatacube SWG](https://www.ogc.org/press-release/ogc-forms-new-geodatacube-standards-working-group/). Essentially, GeoZarr will define a standard for a serverless datacube. Therefore, close coordination between these SWGs seems needed.

Reviewer: ```suggestion
* This SWG is closely related to the newly announced [Geodatacube SWG](https://www.ogc.org/press-release/ogc-forms-new-geodatacube-standards-working-group/). Essentially, Geodatacube will specify a server API while GeoZarr will define a standard for a serverless datacube. Therefore, close coordination between these SWGs seems needed.
```

* xcube Multi-Resolution Datasets: https://github.com/dcs4cop/xcube/blob/master/docs/source/mldatasets.md

=== Details of first meeting

The first meeting of the SWG will occur within four weeks of approval of the SWG charter.

=== Projected on-going meeting schedule

The work of this SWG will be carried out primarily on GitHub and via email, web conferences / calls, and at face-to-face sessions at OGC Member Meetings as agreed to by the SWG members. The web conferences / calls will be scheduled as needed and posted to the OGC portal. Voting on OGC GeoZarr Conventions content will be limited to SWG members only.

=== Supporters of this Charter

The following people support this proposal and are committed to the Charter and projected meeting schedule. These members are known as SWG Founding or Charter members. The charter members agree to the SoW and IPR terms as defined in this charter. The charter members have voting rights beginning the day the SWG is officially formed. Charter Members are shown on the public SWG page.

|===
|Name |Organization

|Christophe Noel | Spacebel
|Brianna R. Pagán | NASA GES DISC
|Alexey N. Shiklomanov | NASA Goddard Space Flight Center

> 

Reviewer 3: Please add |David Blodgett | U.S. Geological Survey

Author: done

> 

Reviewer 8: ```suggestion
|Tyler A. Erickson | Vorgeo
```

== Business Value Proposition

In the geospatial world, new cloud-native data formats are emerging. Zarr is a generic data format for n-dimensional arrays that enables access to data in compressed chunks of the original array and has become increasingly popular to use for geospatial purposes. In June 2022, the OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html). The purpose of this charter is to adopt a more explicit GeoZarr as an OGC Standard which would provide guidance to standardize the approach for encoding various aspects of geospatial data in zarrs.

== Scope of Work

The goal of the GeoZarr specification is to establish flexible and inclusive conventions for the Zarr cloud-native format, specifically designed to meet the diverse requirements within the geospatial domain. These conventions aim to provide a clear and standardized framework for organizing and describing data, ensuring unambiguous representation. 

> The objectives of GeoZarr conventions includes:

Reviewer: ```suggestion
In addition to the encoding geospatial data and metadata using Zarr, the specification will aim to to provide a multidimensional alternative to the two-dimensional Cloud-Optimized GeoTiff format which has gained popularity due to its serverless capabilities. 
 enabling it to inherently support functions traditionally associated with servers. These capabilities allow for inherent support of traditionally server-based functions, including visualization (similar to OGC API Maps), data subset access (analogous to OGC API Coverages), and symbology (equivalent to OGC API Styles). These aspects are planned to be incorporated as optional profiles (e.g. conformance classes).

The objectives of GeoZarr conventions includes:
```

Based on recent discussions, I need a clear commitment confirming that the original purpose of GeoZarr falls within the Working Group's scope: i.e. not only encoding details, but defining standard conventions for map viewing and data analysis clients.

=== Specific Existing Work Used as Starting Point
* GeoZarr draft specification: https://github.com/zarr-developers/geozarr-spec/ 

=== Is This a Persistent SWG

[x] YES

[ ] NO

=== When can the SWG be Inactivated

The SWG can be inactivated once the SWG identifies no new tasks for the SWG and there are no open Change Requests.

== Description of deliverables
The GeoZarr SWG will deliver a candidate Standard and associated developer resources.

The SWG expects to have a candidate Standard ready for OGC Architecture Board (OAB) review and public comment within nine months of creation of the SWG. Because example implementations will be developed at the same time the candidate Standard is formalized, reference implementations that fully use GeoZarr should be documented at the same time the candidate Standard goes to vote.

=== Initial Deliverables

The following deliverables will be the initial results of work of the SWG.

* OGC GeoZarr Standard

* GeoZarr developer resources

> The targeted start date for this SWG is 

Reviewer 8: Is the process close enough to specify a start date? How about "immediately upon approval of the SWG charter"?

== Anticipated Audience / Participants

This SWG will develop a Standard for general use in the geospatial community and suitable for data exchange beyond this community. Geospatial data providers and software implementers will be interested in assisting with the development of this Standard as well as the output of the SWG.

== Domain Working Group Endorsement

The SWG convenors will discuss the charter with potentially interested Domain Working Groups (DWGs) at the first opportunity.

== Other informative information about the work of this SWG

=== Collaboration

All work in the Standards Working Group will be public and the SWG solicits contributions and feedback from OGC members and non-OGC members to the extent that is supported by the OGC Technical Committee Policies and Procedures.

The OGC GeoZarr SWG will collaborate on Standard development using a public GitHub repository and a Gitter channel. Development of the Standard will include the use of Issues and other project tools in GitHub.

=== Similar or Applicable Standards Work (OGC and Elsewhere)

* The OGC endorsed a community standard of Zarr V2.0 (https://zarr.readthedocs.io/en/stable/spec/v2.html) in June 2022.

> * This SWG is closely related to the newly announced [Geodatacube SWG](https://www.ogc.org/press-release/ogc-forms-new-geodatacube-standards-working-group/). Essentially, Geodatacube will specify a server API while GeoZarr will define a standard Cloud-native format for a serverless datacube. Therefore, close coordination between these SWGs seems needed.

Reviewer 8: Suggest "GeoZarr will define" -> "GeoZarr can define" since GeoZarr can be used for data structures other than datacubes (as described earlier in the Mixing Data bullet).

* The XCube project has potential synergies with the GeoZarr specification as it already relies and complies with CF conventions: 

> * xcube Dataset Convention: https://github.com/dcs4cop/xcube/blob/master/docs/source/cubespec.md

Reviewer 8: ```suggestion
  * xcube Dataset Convention: https://github.com/dcs4cop/xcube/blob/master/docs/source/cubespec.md
```

Reviewer 8: Indenting to create a subbullet of the XCube project bullet.

> * xcube Multi-Resolution Datasets: https://github.com/dcs4cop/xcube/blob/master/docs/source/mldatasets.md

Reviewer 8: ```suggestion
  * xcube Multi-Resolution Datasets: https://github.com/dcs4cop/xcube/blob/master/docs/source/mldatasets.md
```

=== Details of first meeting

The first meeting of the SWG will occur within four weeks of approval of the SWG charter.

=== Projected on-going meeting schedule

The work of this SWG will be carried out primarily on GitHub and via email, web conferences / calls, and at face-to-face sessions at OGC Member Meetings as agreed to by the SWG members. The web conferences / calls will be scheduled as needed and posted to the OGC portal. Voting on OGC GeoZarr Conventions content will be limited to SWG members only.

=== Supporters of this Charter

The following people support this proposal and are committed to the Charter and projected meeting schedule. These members are known as SWG Founding or Charter members. The charter members agree to the SoW and IPR terms as defined in this charter. The charter members have voting rights beginning the day the SWG is officially formed. Charter Members are shown on the public SWG page.

|===
|Name |Organization

|Christophe Noel | Spacebel
|Brianna R. Pagán | NASA GES DISC
|Alexey N. Shiklomanov | NASA Goddard Space Flight Center

> |Tyler A. Erickson | Vorgeo

Reviewer 8: ```suggestion
|Tyler A. Erickson | VorGeo
```
